statsmodels/stats/weightstats.py

```diff
@@ -636,6 +636,89 @@ def __init__(self, d1, d2):
 #            d1.nobs1 = d1.sum_weights.astype(float)  #float just to make sure
 #        self.nobs2 = d2.sum_weights.astype(float)
 
+    @classmethod
+    def from_data(cls, data1, data2, weights1=None, weights2=None, 
+            ddof1=0, ddof2=0):
+        '''construct a CompareMeans object from data
+
+        Parameters
+        ----------
+        data1, data2 : array-like, 1-D or 2-D
+            compared datasets
+        weights1, weights2 : None or 1-D ndarray
+            weights for each observation of data1 and data2 respectively,
+            with same length as zero axis of corresponding dataset.
+        ddof1, ddof2 : int
+            default ddof1=0, ddof2=0, degrees of freedom for data1,
+            data2 respectively.
+
+        Returns
+        _______
+        A CompareMeans instance.
+        
+        '''
+        return cls(DescrStatsW(data1, weights=weights1, ddof=ddof1), 
+                DescrStatsW(data2, weights=weights2, ddof=ddof2))
+
+    def summary(self, use_t=True, alpha=0.05, usevar='pooled', value=0):
+        '''summarize the results of the hypothesis test
+
+        Parameters
+        ----------
+
+        use_t : bool, optional
+            if use_t is True, then t test results are returned
+            if use_t is False, then z test results are returned
+        alpha : float
+            significance level for the confidence interval, coverage is
+            ``1-alpha``
+        usevar : string, 'pooled' or 'unequal'
+            If ``pooled``, then the standard deviation of the samples is assumed to be
+            the same. If ``unequal``, then Welsh ttest with Satterthwait degrees
+            of freedom is used
```

Comment:
Because we allow both t-test and z-test here, these could be made more precise,
for example

```
If ``unequal``, then the variance of Welsh ttest will be used, and the degrees of freedom are those of Satterthwaite if ``use_t`` is True.
```

Satterthwaite has "e" at the end